### PR TITLE
feat: collect kubectl information in AKS Windows node log

### PR DESCRIPTION
**Pull request by ShiqianTao**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

Collect kubectl information in AKS Windows node log

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: staging/cse/windows/debug/collect-windows-logs.ps1 [hunk 1]
```diff
@@ -330,6 +330,27 @@ if (Test-Path $nvidiaInstallLogFolder) {
   }
 }
 
+# Collect kubectl information
+Write-Host "Collecting kubectl information"
```
Comment (AbelHu) [inline staging/cse/windows/debug/collect-windows-logs.ps1:334]: Collecting the information of the node and pods?

File: staging/cse/windows/debug/collect-windows-logs.ps1 [hunk 2]
```diff
@@ -330,6 +330,27 @@ if (Test-Path $nvidiaInstallLogFolder) {
...
+function kubectl { c:\k\kubectl.exe --kubeconfig c:\k\config $args }
```
Comment (AbelHu) [inline staging/cse/windows/debug/collect-windows-logs.ps1:335]: We should only collect these info when both `c:\k\kubectl.exe` and `c:\k\config` exist.

File: staging/cse/windows/debug/collect-windows-logs.ps1 [hunk 3]
```diff
@@ -330,6 +330,27 @@ if (Test-Path $nvidiaInstallLogFolder) {
...
+kubectl get nodes -o wide > "$ENV:TEMP\kubectl-get-nodes-$($timeStamp).log"
```
Comment (AbelHu) [inline staging/cse/windows/debug/collect-windows-logs.ps1:336]: We should handle the exception if the command fails, for example, it cannot connect to the api server.

Comment (ShiqianTao) [reply staging/cse/windows/debug/collect-windows-logs.ps1]: Added try-catch, the log is
image

File: staging/cse/windows/debug/collect-windows-logs.ps1 (hunk 2 above)
Comment (ShiqianTao) [reply staging/cse/windows/debug/collect-windows-logs.ps1]: Added check

File: staging/cse/windows/debug/collect-windows-logs.ps1 [hunk 4]
```diff
@@ -330,6 +330,38 @@ if (Test-Path $nvidiaInstallLogFolder) {
...
+    kubectl describe node $nodeName > "$ENV:TEMP\kubectl-describe-nodes-$($timeStamp).log"
+
+    $podsJson = & crictl.exe pods --output json | ConvertFrom-Json
+    foreach ($pod in $podsJson.items) {
+      $podName = $pod.metadata.name
+      $namespace = $pod.metadata.namespace
+      kubectl describe pod $podName -n $namespace >> "$ENV:TEMP\kubectl-describe-pods-$($timeStamp).log"
+    }
+
+    $kubectlLogFiles = Get-ChildItem -Path "$ENV:TEMP\kubectl-*.log"
```
[the remark this answers is not in the document]
Comment (ShiqianTao) [reply staging/cse/windows/debug/collect-windows-logs.ps1:355]: Yes, if we want to exclude older logs, can we remove the `-$($timeStamp)` in the filename?

**Review by AbelHu — APPROVED** [review]
LGTM. Thanks for the PR.

**Pull request merged**